### PR TITLE
Replace brotli.js with native code implementation

### PR DESCRIPTION
**Pull request by WeebDataHoarder**
Removes `brotli.js` and polyfills used to decode _.br_ files within JavaScript.
Instead, use existing `libbrotlidec` on C++ code, using adapted source from [brotli code](https://github.com/google/brotli/blob/7f740f1308336e9ec0afdb9434896307859f5dc9/go/cbrotli/reader.go) (which we depend on).

This saves about 100 KiB or so, mostly extra dictionary size.

### CODE REVIEW

**Review by TheOneric — APPROVED** [review]
Looks good except for a few minor nits.

File: src/brotli.cpp [hunk 1]
```diff
@@ -0,0 +1,95 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdarg.h>
+#include <string.h>
+#include <vector>
+#include <cstdint>
+
+#include "../lib/brotli/c/include/brotli/decode.h"
```
[the remark this answers is not in the document]
Comment (WeebDataHoarder) [reply src/brotli.cpp:8]: I'll make this change, then wouldn't `#include "../lib/libass/libass/ass.h"` on _libass.cpp_ / _SubtitleOctopus.cpp_ be the same? I'll check if that is correct and make another minor PR

File: src/brotli.cpp [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+//Adjusted from brotli go creader
+//
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Distributed under MIT license.
+// See file LICENSE for detail or copy at https://opensource.org/licenses/MIT
```
[the remark this answers is not in the document]
Comment (WeebDataHoarder) [reply src/brotli.cpp]: This has been moved to the top of _brotli.cpp_

File: src/post-worker.js [hunk 3]
```diff
@@ -103,14 +103,46 @@ self.freeTrack = function () {
     self.getRenderMethod()();
 };
 
+self.decodeBrotliArray = function (uint8array) {
+    //Allocate and copy brotli encoded input into the heap, using buf ptr as offset
+    var buf = Module._malloc(uint8array.length);
+    Module["HEAPU8"].set(uint8array, buf);
+
+    //Allocate return length pointer
```
[the remark this answers is not in the document]
Comment (WeebDataHoarder) [reply src/post-worker.js]: Altered comment to use that wording.

**Comment by WeebDataHoarder** [comment]
Applied the changes suggested on the review.

File: src/brotli.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (TheOneric) [reply src/brotli.cpp]: Done.

**Pull request merged**